usr/sbin/cloudguest-lic-watcher

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/python3
+
+# Copyright (c) 2025, SUSE LLC, All rights reserved.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3.0 of the License, or (at your option) any later version.
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.
+
+import logging
+import os
+
+import instance_billing_flavor_check.utils as inst_flvr_utils
+import cloudregister.registerutils as utils
+
+CACHE_LICENSE_PATH = os.path.join(utils.get_state_dir(), 'cached_license')
+
+def update_license_cache(license_type):
+    """Update the cache to track the license type"""
+    with open(CACHE_LICENSE_PATH, 'w+') as cached_license:
+        cached_license.write(license_type)
+
+def has_license_changed(license_type):
+    """Check license type changes against metadata info"""
+    try:
+        with open(CACHE_LICENSE_PATH, 'r') as cached_license:
+            old_license = cached_license.read()
```

Comment:
no lstrip() needed ? no CR or other space characters that could harm the subsequent comparison expected ?

Reply:
We control the write operation, I think we can do without normalizing what we read.